src/PlatformBenchmarks/BenchmarkApplication.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
+
+namespace PlatformBenchmarks
+{
+    public class BenchmarkApplication : HttpConnection
+    {
+        private static AsciiString _crlf = "\r\n";
+        private static AsciiString _http11OK = "HTTP/1.1 200 OK\r\n";
+        private static AsciiString _headerServer = "Server: Custom";
+        private static AsciiString _headerContentLength = "Content-Length: ";
+        private static AsciiString _headerContentLengthZero = "Content-Length: 0";
+        private static AsciiString _headerContentTypeText = "Content-Type: text/plain";
+
+        private static readonly DateHeaderValueManager _dateHeaderValueManager = new DateHeaderValueManager();
+
+        private static AsciiString _plainTextBody = "Hello, World!";
+
+        private static class Paths
+        {
+            public static AsciiString Plaintext = "/plaintext";
+        }
+
+        private bool _isPlainText;
+
+        public override void OnStartLine(HttpMethod method, HttpVersion version, Span<byte> target, Span<byte> path, Span<byte> query, Span<byte> customMethod, bool pathEncoded)
+        {
+            _isPlainText = method == HttpMethod.Get && path.StartsWith(Paths.Plaintext);
+        }
+
+        public override void OnHeader(Span<byte> name, Span<byte> value)
+        {
+        }
+
+        public override ValueTask ProcessRequestAsync()
+        {
+            if (_isPlainText)
+            {
+                PlainText(Connection.Transport.Output);
+            }
+            else
+            {
+                Default(Connection.Transport.Output);
+            }
+
+            return default;
+        }
+
+        public override async ValueTask OnReadCompletedAsync()
+        {
+            await Connection.Transport.Output.FlushAsync();
+        }
+
+        private static void Default(PipeWriter pipeWriter)
+        {
+            var writer = new BufferWriter<PipeWriter>(pipeWriter);
+
+            // HTTP 1.1 OK
+            writer.Write(_http11OK);
+
+            // Server headers
+            writer.Write(_headerServer);
+
+            // Date header
+            writer.Write(_dateHeaderValueManager.GetDateHeaderValues().Bytes);
+            writer.Write(_crlf);
+
+            // Content-Length 0
+            writer.Write(_headerContentLengthZero);
```

Comment:
This constant could have CRLF at the end so you dont have to write twice.

Reply:
Nice catch!